File: My Project-2_noise/plot_audio.m
clear;
[y,fs] = audioread('Test-Data/Adhiraj_1.wav');
[y1,fs1]= audioread('Train-Data/Adhiraj-speech/Adhiraj-1.wav');
t = 0:1/fs:(length(y)-1)/fs;
t1 = 0:1/fs1:(length(y1)-1)/fs1;
subplot(4,2,1);
plot(t1,y1(:,1));
xlabel('time','FontWeight','bold');
ylabel('audio signal','FontWeight','bold');
title('Train Audio without noise');
grid on;
A = [6,4,2];
j = 3;

for i = A
    subplot(4,2,j);
    n = sqrt(10^(-i)).*randn(size(y1));
    y2 = y1 + n;
    plot(t1,y2(:,1));
    grid on;
    xlabel('time','FontWeight','bold');
    ylabel('audio signal','FontWeight','bold');
    title(strcat('Train Audio with noise of variance 10^{-',num2str(i),'}'));
    j = j+2;
end

subplot(4,2,2);
plot(t,y(:,1));
xlabel('time','FontWeight','bold');
ylabel('audio signal','FontWeight','bold');
title('Test Audio without noise');
grid on;
A = [6,4,2];
j = 4;

for i = A
    subplot(4,2,j);
    n = sqrt(10^(-i)).*randn(size(y));
    y2 = y + n;
    plot(t,y2(:,1));
    grid on;
    xlabel('time','FontWeight','bold');
    ylabel('audio signal','FontWeight','bold');
    title(strcat('Test Audio with noise of variance 10^{-',num2str(i),'}'));
    j = j+2;
end

    

File: My Project-2_noise/plot_audio.py
import pickle as cPickle
import matplotlib.pyplot as plt
import numpy as np
from scipy.signal import wiener
import librosa as lib
from sklearn import mixture
from feature_extr import extract_features
import warnings
warnings.filterwarnings("ignore")

source   = "Train-Data/" #Here, the training data is saved
dest = "Sp_Models/"
train_file = "Traindatapath.txt" #Training data path is stored in text file       
file_paths = open(train_file,'r')


audio,sr = lib.load(source + 'Adhiraj-speech/Adhiraj-1.wav')
plt.subplot(2,2,1)
plt.plot(audio)
plt.xlabel('time')
plt.ylabel('audio signal')
plt.title('audio without noise')
j = 2
A = [8,6,4]
for i in A:
    noise = np.random.normal(0,np.sqrt(pow(10,-i)),audio.shape)
    new_audio = audio + noise
    plt.subplot(2,2,j)
    plt.plot(new_audio)
    plt.xlabel('time')
    plt.ylabel('audio signal')
    plt.title('audio with noise of variance 10^-'+str(i))
    j+=1
plt.show()
